03a_it_analysis/src/estuary_salinity_functional_performance_func.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Oct 18 12:54:24 2022
+
+@author: ggorski
+"""
+import it_functions
+import itertools
+import numpy as np
+import os
+import pandas as pd
+import xarray as xr
+
+
+def find_reps(run_id):
+    '''
+    find the list of directories that house the replicates
+    Parameters
+    ----------
+    run_id : str
+        the run number to aggregate
+
+    Returns
+    -------
+    reps : list
+        list of replicates
+
+    '''
+    reps = list()
+    rootdir = os.path.join('03b_model/out',run_id)
+    for file in os.listdir(rootdir):
+        d = os.path.join(rootdir, file)
+        #replicates have to be in a directory with a two character name(e.g., "02")
+        if os.path.isdir(d) and len(file) == 2:
+            reps.append(file)
+    return reps
+
+def create_cleaned_io_file(run_id):
+    '''
+    creates a cleaned csv with aggregate results from 5 replicates and merges
+    with input data and outputs from COAWST model, saves csv to 03b_model/out
+    directory
+
+    Parameters
+    ----------
+    run_id : str
+        the run number to aggregate
+
+    Returns
+    -------
+    None.
+
+    '''
+    #reps = ['00','01','02','03','04']
+    reps = find_reps(run_id)
+    ml_sf = pd.read_csv(os.path.join('03b_model/out',run_id,reps[0],'ModelResults.csv'), parse_dates = True, index_col = 'Unnamed: 0')
+    for i,rep in enumerate(reps):
+        temp = pd.read_csv(os.path.join('03b_model/out',run_id,rep,'ModelResults.csv'), parse_dates = True, index_col = 'Unnamed: 0')
+        ml_sf['saltfront_pred_'+rep] = temp['saltfront_pred'].copy()
+
+    ml_sf_mean = ml_sf[['saltfront_pred','saltfront_pred_01','saltfront_pred_02','saltfront_pred_03','saltfront_pred_04']].mean(axis = 1)
```

Comment:
This also needs adjusting for number of reps, right?

Reply:
Yep! Great point, tested it with 5 replicates and it worked. Ok should be ready now

Reply:
okay, you can merge